hare/hare.go

```diff
@@ -164,12 +172,8 @@ func (h *Hare) collectOutput(output TerminationOutput) error {
 	}
 
 	h.mu.Lock()
-	if len(h.outputs) == h.bufferSize {
-		for k := range h.outputs {
-			if h.isTooLate(instanceId(k)) {
-				delete(h.outputs, k)
-			}
-		}
+	if len(h.outputs) >= h.bufferSize {
+		delete(h.outputs, h.oldestResultInBuffer())
 	}
 	h.outputs[types.LayerID(id)] = blocks
```

Comment:
shouldn't we ask if it's too late before we add to buffer?
(I'm not sure that we should)

Reply:
if you look a few lines above you'll see that we're checking it beforehand.